src/systems/filecoin_vm/runtime/runtime.go

```diff
@@ -522,19 +521,6 @@ func (rt *VMContext) AcquireState() ActorStateHandle {
 	}
 }
 
-func (rt *VMContext) VerifySignature(signerActor addr.Address, sig filcrypto.Signature, m filcrypto.Message) bool {
-	st := rt._globalStatePending.Impl().GetActorState(signerActor)
-	if st == nil {
-		rt.AbortAPI("VerifySignature: signer actor not found")
-	}
-	pk := st.GetSignaturePublicKey()
-	if pk == nil {
-		rt.AbortAPI("VerifySignature: signer actor has no public key")
-	}
-	ret := rt.Compute(ComputeFunctionID_VerifySignature, []Any{pk, sig, m})
-	return ret.(bool)
-}
```

Comment:
[Blocking] I think we need this method. Actors clearly need to verify each other's signatures, and we don't want to do this by sending a message to the originating actor asking it to verify its own signature. Maybe it could just abort if the address provided isn't an AccountActor?

Reply:
(please consider the PR approved conditional on this -- GitHub doesn't seem to let me update the review state to reflect that)

Reply:
> Actors clearly need to verify each other's signatures

When? I removed this runtime call without any actor code breaking. I don't think we should specify runtime methods that are not used. What's the use case for in-actor signature verification?
- message and block signatures are checked prior to evaluation, assumed correct within the VM (which can't see the signatures)
- tickets and election post are verified outside actor code

There could well be something, but we should know what it is before implementing this (correctly).

Reply:
I'm ok with removing this for now, since we're refactoring the checking of actor principals anyway. Ideally with the new design we won't require any signature verifications, since the origin of a message (as supplied by the VM context) is sufficient authentication. There might be some corner cases we're missing, though.